.gitignore

```diff
@@ -4,10 +4,11 @@
 /.idea/
 /target/
 node_modules/
+package-lock.json
```

Comment:
On this branch, I decided to start doing many small commits to see if I liked that type of workflow.  I did since it was more convenient to then pick and choose which parts of my work would go into the final version for the PR.  So S my D, B.